orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -1097,6 +1097,67 @@ Example 4:
         }
       });
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(ReduceSumTraining)
+      .SetDomain(kOnnxDomain)
+      .SinceVersion(12)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+      .SetDoc("ReduceSumTraining")
+      .Attr("keepdims",
+            "Keep the reduced dimension or not, default 1 mean keep reduced dimension.",
+            AttributeProto::INT,
+            static_cast<int64_t>(1))
+      .AllowUncheckedAttributes()
+      .Input(0, "data", "An input tensor.", "T")
+      .Input(1, "axes",
+             "A list of integers, along which to reduce. The default is to reduce over "
+             "all the dimensions of the input tensor. Accepted range is [-r, r-1] where r = rank(data).",
+             "tensor(int64)")
+      .Output(0, "reduced", "Reduced output tensor.", "T")
+      .TypeConstraint(
+          "T",
+          OpSchema::numeric_types_for_math_reduction(),
+          "Constrain input and output types to high-precision numeric tensors.")
+      .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+
+        int64_t keep_dims = 1;
+        auto attr_proto = ctx.getAttribute("keepdims");
+        if (attr_proto) {
+          keep_dims = attr_proto->i();
+        }
+        auto& input_shape = ctx.getInputType(0)->tensor_type().shape();
+        int64_t input_ndim = input_shape.dim_size();
+        auto output_shape =
+            ctx.getOutputType(0)->mutable_tensor_type()->mutable_shape();
+        std::vector<int64_t> axes;
+        auto axes_proto = ctx.getInputData(1);
+        if (axes_proto) {
+          axes = ParseData<int64_t>(axes_proto);
+        }        
+        
+        for (size_t i = 0; i < axes.size(); ++i) {
+          if (axes[i] < 0)
+            axes[i] += input_ndim;
+        }
+        // do we need handle negative axis?
```

Comment:
yes.

Reply:
Per discussion w/ Aishwarya, we may disable the inference logic.

---
In reply to: [455329194](https://github.com/microsoft/onnxruntime/pull/4522#discussion_r455329194) [](ancestors = 455329194)